src/declarativewebpage.cpp

```diff
@@ -311,10 +299,17 @@ QString DeclarativeWebPage::saveToFile(QImage image, QRect cropBounds)
 
     // 75% quality jpg produces small and good enough capture.
     QString path = QString("%1/tab-%2-thumb.jpg").arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation)).arg(m_tabId);
```

Comment:
Not racing over here. :)

SetTabId is called when a webpage is created. The tabId remains the same during the life time of the page. If browser kills a tab due to out-of-memory or so that maximum number of tabs is reached, webpage is destroyed. However, when a tab is resurrected (accessed again) a new webpage is created and a tab id assigned back for the page.

The setTabId exists due to the code paste below (webpages.cpp 115-140, shortened code clip):

```
QObject *object = m_webPageComponent->beginCreate(context);
webPage = qobject_cast<DeclarativeWebPage *>(object);
webPage->setTabId(tabId);
m_webPageComponent->completeCreate();
```
We could move DeclarativeWebPage::setTabId to private and make WebPages to be a friend class. But I don't see that as relevant change for now.